content/en/docs/architecture/step-registry.md

```diff
@@ -959,3 +959,59 @@ tests:
   steps:
     workflow: openshift-e2e-disruptive
 {{< / highlight >}}
+
+## VPN connection
+
+For platforms that need access to restricted environments, `ci-operator`
+supports adding a dedicated VPN connection to each test step.  Since this is a
+requirement of specific platforms, it is triggered by special files in the
+[cluster profile(s)]({{< ref "docs/how-tos/adding-a-cluster-profile#what-is-a-cluster-profile" >}})
+associated with those platforms.  This process is transparent to the test
+command: when a VPN connection is requested at the test level, it is set up
+automatically by the test platform with no changes required to individual tests.
+
+{{< alert title="Note" color="info" >}}
+Details of the implementation can be found in the
+<a href="https://docs.google.com/document/d/1mPjrHVS1EvmLdq4kGhRazTpGu6xVZDyGpVAphVZhX4w/edit?resourcekey=0-KA-qXXq1J2bTR7o6Kit9Vw">design document</a>.
+{{< /alert >}}
+
+### Cluster profile
+
+VPN connections are requested by the presence of special files in the cluster
+profile, detected when the test steps are about to be executed:
+
+- `vpn_image.txt`: _pull spec_ of the image to be used for the VPN client
+  container.  It should contain the files necessary to establish the connection
+  as well as the `bash` shell.
+- `vpn.sh`: entry point of the container, executed effectively as `bash -c
+  "$(<"${type.sh}")"`.
+
+### Client container
+
+The container should execute the VPN client to establish the connection.  It
+will reside in the same network namespace as the test container, so no special
+provisions are necessary to make the networking changes usable by the test
+program.  When executed, the entry point program will be located in the
+directory where the cluster profile files are mounted, so all secrets will be
+available and can be referenced with a relative path.
+
+In addition to executing the client, the container should also watch for the
+creation of a file indicating that the main test has finished and then exit so
+that the test can terminate properly.  This marker file is created automatically
+by the CI infrastructure at `/logs/marker-file.txt`.  The client can perform
+these actions with a script such as:
+
+{{< highlight bash >}}
+function exit_with_test() {
+    until [[ -f /logs/marker-file.txt ]]; do sleep 1; done
+    while :; do kill 1; sleep 1; done
+}
+
+exit_with_test &
+# run the VPN client
+{{< / highlight >}}
+
+{{< alert title="Note" color="info" >}}
+N.b. both containers start simultaneously, so the test may exit before the VPN
```

Comment:
What should the test container do?

- wait long enough to start to ensure the VPN is connected?
- watch a file generated by the VPN container, similar to `/logs/marker-file.txt` (only with opposite generator and consumer).

Reply:
That is highly dependent on the test steps, VPN clients, _and_ network configuration, none of which we control.

The only general observable effect is the network connection itself, I can't think of a way we could provide this signal in the test platform.

Reply:
The client script requirement is now described in the first list item in the section.

The test should not be concerned with this, so it's vaguely described in the new section.